### PR TITLE
MAINT: Cleanup Python2 compatibility code

### PR DESCRIPTION
**Pull request by sethtroisi**
#### Reference issue

Continued work to remove all Python2 Code #11584

Follows after #11760

#### What does this implement/fix?

Removes use of numpy py3k.h code.

### CODE REVIEW

File: scipy/io/matlab/streams.pyx [hunk 1]
```diff
@@ -19,14 +19,6 @@ cdef extern from "Python.h":
         pass
     ctypedef struct FILE
 
-cdef extern from "py3k.h":
-    FILE* npy_PyFile_Dup(object file, char *mode) except NULL
-    int npy_PyFile_DupClose(object file, FILE *handle) except -1
-    int npy_PyFile_Check(object file)
-
-
-cdef bint IS_PYPY = ('__pypy__' in sys.modules)
```
Comment (tylerjereddy) [inline scipy/io/matlab/streams.pyx:28]: Can you clarify why removing this `pypy`-related code make sense in the Python 2 maintenance cleanup?

I'm not saying it is wrong to remove it, but we don't currently actively test `pypy` in CI (though we used to) and another reviewer may wonder the same when they are looking through this PR.

Comment (sethtroisi) [reply scipy/io/matlab/streams.pyx:28]: This line should have been removed in https://github.com/scipy/scipy/pull/11655

A quick grep suggests that no uses of `IS_PYPY` in c code, in python it's defined and used in several unrelated places (and defined but not used in `scipy/io/matlab/tests/test_streams.py`)

Comment (tylerjereddy) [reply scipy/io/matlab/streams.pyx:28]: Yeah, I saw it used in a few places with `git grep` so wasn't clear on it

Comment (tylerjereddy) [reply scipy/io/matlab/streams.pyx:28]: But, strictly speaking, that' s not related to the Python 2 stuff, right?

Comment (sethtroisi) [reply scipy/io/matlab/streams.pyx:28]: it as used by a Python2 only method so IDK, but it's a non issue because I moved it to the other PR.

**Comment by rgommers** [comment]
Merged, thanks @sethtroisi. Completely deleting a `py3k.h` feels good:)

**Pull request merged**